odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,173 @@
+    operator's, and ``h`` is a convex and differentiable functional.
+
+    The method can also be used to solve the more general problem
+
+        ``min_x f(x) + sum_i (g_i @ l_i)(L_i x)``,
+
+    where ``l_i`` are convex functionals and @ is the infimal convolution:
+
+        ``(f @ g)(x) = inf_y { f(y) + g(x-y) }``.
+
+    Parameters
+    ----------
+    x : `LinearSpaceVector`
+        Initial point, updated in place.
+    prox_f : `callable`
+        `Proximal factory` for the functional ``f``.
+    prox_cc_g : `sequence` of `callable`'s
+        Sequence of `proximal factories` for the convex conjuates of the
+        functionals ``g_i``.
+    grad_h : `Operator`
+        Operator representing the gradient of  ``h``.
+    L : `sequence` of `Operator`
+        A sequence with as many elements as ``prox__cc_gs`` of linear operators
+        ``L_i``.
+    tau : `float`
+        Step size-like parameter for ``prox_f``.
+    sigma : `sequence` of  `float`
+        Sequence of step size-like parameters for the sequence ``prox_cc_g``.
+    niter : `int`
+        Number of iterations.
+    callback : `Callback`, optional
+        Show partial results.
+
+    Other Parameters
+    ----------------
+    grad_cc_l : `sequence` of `Operator`, optional
+        Sequence of operators representing the gradients of  ``l_i^*``.
+        If omitted, the simpler problem without ``l_i``  will be considered.
+
+    Notes
+    -----
+    The exact conditions on the involved functionals are as follows: :math:`f`
+    and :math:`g_i` are proper, convex and lower semicontinuous, and :math:`h`
+    is convex and differentialbe with :math:`\\eta^{-1}`-Lipschitz continuous
+    gradient, :math:`\\eta > 0`.
+
+    The optional operators :math:`\\nabla l_i^*` need to be
+    :math:`\\nu_i^{-1}`-Lipschitz continuous. Note that in the reference
+    [BC2015]_, the condition is formulated as :math:`l_i` being proper, lower
+    semicontinuous, and :math:`\\nu_i^{-1}`-strongly convex, which implies that
+    :math:`l_i^*` have :math:`\\nu_i`-Lipschitz continuous gradients.
+
+    If the optional operators :math:`\\nabla l_i^*` are omitted, the simpler
+    problem without :mat:`l_i` will be considered. Mathematically, this is done
+    by taking :math:`l_i` to be the functionals that are zero only in the zero
+    element and :math:`\\infty` otherwise. This gives that :math:`l_i^*` are
+    the zero functionals, and hence the corresponding gradients are the zero
+    operators.
+
+    To guarantee convergence, the parameters ``tau``, ``sigma`` and
+    ``L`` need to satisfy
+
+    .. math::
+
+       2 \min \{ \\frac{1}{\\tau}, \\frac{1}{\sigma_1}, \\ldots,
+       \\frac{1}{\sigma_m} \} \cdot \min\{ \\eta, \\nu_1, \\ldots, \\nu_m  \}
+       \cdot \\sqrt{1 - \\tau \\sum_{i=1}^n \\sigma_i ||L_i||^2} > 1,
+
+    where, if the simpler problem is considered, all :math:`\\nu_i` can be
+    considered to be :math:`\\infty`.
+
+
```

Comment:
Too many lines and weird line breaks because this was supposed to be removed... I missed to do that. (This is wrong, and when I harmonized the notation with the notation in the article I realized that).